src/environments/array.js

```diff
@@ -45,18 +45,22 @@ type ArrayEnvNodeDataIncomplete = {|
     // Before these fields are filled.
     body?: AnyParseNode[][],
     rowGaps?: (?ParseNode<"size">)[],
-    numHLinesBeforeRow?: number[],
+    hLinesBeforeRow?: Array<boolean[]>,
 |};
 
-function getNumHLines(parser: Parser): number {
-    let n = 0;
+function getHLines(parser: Parser): boolean[] {
+    // Return an array. The array length = number of hlines.
+    // Each element in the array tells if the line is dashed.
+    const hlineInfo = [];
     parser.consumeSpaces();
-    while (parser.nextToken.text === "\\hline") {
+    let nxt = parser.nextToken.text;
+    while (nxt === "\\hline" || nxt === "\\hdashline") {
         parser.consume();
-        n++;
+        hlineInfo.unshift(nxt === "\\hdashline");
```

Comment:
Why is this `unshift` instead of `push`? In particular, why is the order reversed?

Efficiency wise, it'd be better to `push` and then reverse e.g. while returning. Though I can't inagine anyone having more than a few consecutive `\hline`s.